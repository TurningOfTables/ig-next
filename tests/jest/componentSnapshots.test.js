import { render } from "@testing-library/react";
import GameCards from "../../components/gameCards";
import SharedHead from "../../components/sharedHead";
import Footer from "../../components/footer";
import FeatureSection from "../../components/featureSection";
import GameCard from "../../components/gameCard";
import SteamWidget from "../../components/steamWidget";
import SteamBanner from "../../components/steamBanner";
import GameScreenshot from "../../components/gameScreenshot";
import NavBar from "../../components/navBar";
import ContactUs from "../../components/contactUs";
import GameDetails from "../../components/gameDetails";
import GameDetailsSection from "../../components/gameDetailsSection";
import PresskitSection from "../../components/presskitSection";
import Video from "../../components/video";

it("renders <ContactUs /> component unchanged", () => {
  const { container } = render(<ContactUs />);
  expect(container).toMatchSnapshot();
});

it("renders <FeatureSection /> component unchanged", () => {
  const { container } = render(
    <FeatureSection
      title="Test Title"
      text="Test Text"
      imgSrc="/tmlCard-1.png"
    />
  );
  expect(container).toMatchSnapshot();
});

it("renders <FeatureSection /> reversed component unchanged", () => {
  const { container } = render(
    <FeatureSection
      title="Test Title"
      text="Test Text"
      imgSrc="/tmlCard-1.png"
      reversed
    />
  );
  expect(container).toMatchSnapshot();
});

it("renders <Footer /> component unchanged", () => {
  const { container } = render(<Footer />);
  expect(container).toMatchSnapshot();
});

it("renders <GameCard /> component unchanged", () => {
  const { container } = render(
    <GameCard
      link="/test"
      steamStoreId="0"
      imagePath="bg-[url('/tmlCard.png')]"
    />
  );
  expect(container).toMatchSnapshot();
});

it("renders <GameCards /> component unchanged", () => {
  const { container } = render(<GameCards />);
  expect(container).toMatchSnapshot();
});

it("renders <GameDetails /> component unchanged", () => {
  const { container } = render(<GameDetails />);
  expect(container).toMatchSnapshot();
});

it("renders <GameDetailsSection /> component unchanged", () => {
  const { container } = render(
    <GameDetailsSection title="Test" content="Test" />
  );
  expect(container).toMatchSnapshot();
});

it("renders <GameDetailsSection /> store link component unchanged", () => {
  const { container } = render(
    <GameDetailsSection title="Test" content="Test" storeLink />
  );
  expect(container).toMatchSnapshot();
});

it("renders <GameScreenshot /> component unchanged", () => {
  const { container } = render(
    <GameScreenshot title="Test" imgSrc="/test.png" />
  );
  expect(container).toMatchSnapshot();
});

it("renders <PresskitSection /> component unchanged", () => {
  const { container } = render(<PresskitSection title="Test" content="Test" />);
  expect(container).toMatchSnapshot();
});

it("renders <SharedHead /> component unchanged", () => {
  const { container } = render(<SharedHead />);
  expect(container).toMatchSnapshot();
});

it("renders <SteamBanner /> component unchanged", () => {
  const { container } = render(<SteamBanner />);
  expect(container).toMatchSnapshot();
});

it("renders <SteamWidget /> component unchanged", () => {
  const { container } = render(<SteamWidget />);
  expect(container).toMatchSnapshot();
});

it("renders <Video /> component unchanged", () => {
  const { container } = render(<Video videoId="8UPMQbm8WkQ" title="Test" />);
  expect(container).toMatchSnapshot();
});
