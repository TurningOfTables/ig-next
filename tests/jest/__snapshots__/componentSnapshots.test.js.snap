// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <FeatureSection /> component unchanged 1`] = `
<div>
  <div>
    <div
      class="flex flex-wrap m-10 justify-center"
      data-testid="feature-section-test-title"
    >
      <div
        class="flex flex-col text-center max-w-2xl my-auto"
      >
        <h2
          class="text-2xl"
        >
          Test Title
        </h2>
        <p
          class="m-5"
        >
          Test Text
        </p>
      </div>
      <div
        class="flex flex-col my-auto"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27270%27%20height=%27270%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Test Title image"
            data-nimg="intrinsic"
            data-testid="feature-section-test-title-image"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="flex flex-row mx-auto mt-2 cursor-pointer"
        >
          <div
            class="mx-1"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Zoom icon"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <small>
            Zoom
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders <Footer /> component unchanged 1`] = `
<div>
  <footer
    class="text-left pl-3 mt-2 bottom-0 bg-black text-white fixed w-full"
    data-testid="footer"
  >
    Â© Inexplicable Games
  </footer>
</div>
`;

exports[`renders <GameCards /> component unchanged 1`] = `
<div>
  <div
    class="flex flex-col text-4xl mt-20"
  >
    <h1>
      Projects
    </h1>
    <div
      class="flex flex-row flex-wrap max-w-3xl mx-auto my-5 justify-center"
    >
      <a
        href="/showrunner"
      >
        <div
          class="flex justify-center m-5 w-72 h-72 bg-cover hover:shadow-2xl transition ease-in-out hover:scale-105 duration-200 bg-[url('/showrunnerCard.png')]"
        />
      </a>
      <a
        href="/this-merchant-life"
      >
        <div
          class="flex justify-center m-5 w-72 h-72 bg-cover hover:shadow-2xl transition ease-in-out hover:scale-105 duration-200 bg-[url('/tmlCard.jpg')]"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`renders <SharedHead /> component unchanged 1`] = `<div />`;
